src/super_gradients/training/utils/detection_utils.py

```diff
@@ -242,18 +242,26 @@ def box_area(box):
     return inter / (area1[:, None] + area2 - inter)  # iou = inter / (area1 + area2 - inter)
 
 
-def non_max_suppression(prediction, conf_thres=0.1, iou_thres=0.6, multi_label_per_box: bool = True, with_confidence: bool = False):
+def non_max_suppression(
+    prediction, conf_thres=0.1, iou_thres=0.6, multi_label_per_box: bool = True, with_confidence: bool = False, class_agnostic_nms: bool = False
+):
     """
     Performs Non-Maximum Suppression (NMS) on inference results
-        :param prediction: raw model prediction. Should be a list of Tensors of shape (cx, cy, w, h, confidence, cls0, cls1, ...)
-        :param conf_thres: below the confidence threshold - prediction are discarded
-        :param iou_thres: IoU threshold for the nms algorithm
-        :param multi_label_per_box: whether to use re-use each box with all possible labels
-                                    (instead of the maximum confidence all confidences above threshold
-                                    will be sent to NMS); by default is set to True
-        :param with_confidence: whether to multiply objectness score with class score.
-                                usually valid for Yolo models only.
-        :return: detections with shape nx6 (x1, y1, x2, y2, object_conf, class_conf, class)
+
+    :param prediction: raw model prediction. Should be a list of Tensors of shape (cx, cy, w, h, confidence, cls0, cls1, ...)
+    :param conf_thres: below the confidence threshold - prediction are discarded
+    :param iou_thres: IoU threshold for the nms algorithm
+    :param multi_label_per_box: whether to use re-use each box with all possible labels
```

Comment:
The class_agnostic_nms alongside this way of phrasing multi_label_per_box docs might be a bit confusing.
To me this is a bit more clear:
multi_label_per_box: When True, a box with any class confidences > conf_thres will be passed to the NMS algorithm as multiple boxes (with the same coordinates) with different labels. When False, only a single box will be passed to the NMS algorithm with the label of the highest score.

Reply:
Fixed in ff370f0995797c53b43996b7570bca694a2264dd